pkg/sql/logictest/testdata/logic_test/array

```diff
@@ -1016,3 +1016,63 @@ query T
 SELECT ARRAY_POSITIONS(NULL::STRING[], 'A')
 ----
 NULL
+
+query T
+SELECT STRING_TO_ARRAY('axbxc', 'x')
+----
+{"a","b","c"}
+
+query T
+SELECT STRING_TO_ARRAY('~a~~b~c', '~')
+----
+{"","a","","b","c"}
+
+query T
+SELECT STRING_TO_ARRAY('~foo~~bar~baz', '~', 'bar')
+----
+{"","foo","",NULL,"baz"}
+
+query T
+SELECT STRING_TO_ARRAY('xx~^~yy~^~zz', '~^~', 'yy')
+----
+{"xx",NULL,"zz"}
+
+query T
+SELECT STRING_TO_ARRAY('foo', '')
+----
+{"foo"}
+
+query T
+SELECT STRING_TO_ARRAY('', '')
+----
+{}
+
+query T
+SELECT STRING_TO_ARRAY('', 'foo')
+----
+{}
+
+query T
+SELECT STRING_TO_ARRAY('a', NULL)
+----
+{"a"}
+
+query T
+SELECT STRING_TO_ARRAY(NULL, 'a')
+----
+NULL
+
+query T
+SELECT STRING_TO_ARRAY(NULL, 'a', 'b')
+----
+NULL
+
+query T
+SELECT STRING_TO_ARRAY('a', 'foo', NULL)
+----
+{"a"}
+
+query T
+SELECT STRING_TO_ARRAY('foofoofoofoo', 'foo', 'foo')
+----
+{"","","","",""}
```

Comment:
like we talked about offline, need test cases (and probably matching functionality :) for the empty string delimiter, which is supposed to just return the entire input as a single value in an array.